clippy_lints/src/methods/append_instead_of_extend.rs

```diff
@@ -0,0 +1,41 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::ty::{is_type_diagnostic_item, is_type_lang_item};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, LangItem};
+use rustc_lint::LateContext;
+use rustc_span::symbol::sym;
+
+use super::APPEND_INSTEAD_OF_EXTEND;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, arg: &Expr<'_>) {
+    if_chain! {
+        if let ty = cx.typeck_results().expr_ty(recv).peel_refs();
```

Comment:
That works, but you should put this in front of the `if_chain! {`. So:
```suggestion
    let ty = cx.typeck_results().expr_ty(recv).peel_refs();
    if_chain! {
```

Same for the `if let mut applicability` below, which should be moved into the `then { }` block.

---

Unrelated: I'm surprised Clippy doesn't have a lint for irrefutable bindings in `if let` expressions.